front/jest.config.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * For a detailed explanation regarding each configuration property and type check, visit:
+ * https://jestjs.io/docs/configuration
+ */
+
+export default {
+  testEnvironment: 'jsdom',
+  // All imported modules in your tests should be mocked automatically
+  // automock: false,
+
+  // Stop running tests after `n` failures
+  // bail: 0,
+
+  // The directory where Jest should store its cached dependency information
+  // cacheDirectory: "C:\\Users\\Professional\\AppData\\Local\\Temp\\jest",
+
+  // Automatically clear mock calls, instances, contexts and results before every test
+  clearMocks: true,
+
+  // Indicates whether the coverage information should be collected while executing the test
+  collectCoverage: true,
+
+  // An array of glob patterns indicating a set of files for which coverage information should be collected
+  collectCoverageFrom: ['src/**/*.{ts,tsx}'],
+
+  // The directory where Jest should output its coverage files
+  coverageDirectory: "coverage",
+
+  // An array of regexp pattern strings used to skip coverage collection
+  // coveragePathIgnorePatterns: [
```

Comment:
> why is it commented out?

@skyboyer 
A few reasons
  - it was commented by default, right after command "jest --init"
  - for me everything works well without it uncommented
  - I tried to uncomment it and nothing changed. For example, the folder "node_modules" still is not in the code coverage report and the test time didn't change.
  
In my opinion either specify an option "collectCoverageFrom" or "coveragePathIgnorePatterns". I specified the first one.
Should I delete all commented code in jest.config.ts?

Reply:
my bad, since there is `  collectCoverageFrom: ['src/**/*.{ts,tsx}'],` above, node_modules will be excluded

Reply:
I made several tests on my local machine with my current jest.config.ts state and found out two things:

1. If I uncomment `coveragePathIgnorePatterns: [ "\\\\node_modules\\\\" ]` neither tests execution time changes nor a coverage report
2. If I leave commented `coveragePathIgnorePatterns` and additionally comment `collectCoverageFrom: ['src/**/*.{ts,tsx}']` then tests execution time slightly decreases (about 0.8 second). That happens because now a coverage report doesn't include modules without tests for them. Frankly, I expected the opposite, I supposed that it would take much more time and would include every folder in my project including "node_modules"  

![test_coverage_without_collectCoverageFrom](https://user-images.githubusercontent.com/23404373/203355285-98546c75-7079-4508-a310-cfa40c4d3262.jpg)

Reply:
@skyboyer 
I made several tests on my local machine with my current jest.config.ts state and found out two things:

1. If I uncomment `coveragePathIgnorePatterns: [ "\\\\node_modules\\\\" ]` neither tests execution time changes nor a coverage report
2. If I leave commented `coveragePathIgnorePatterns` and additionally comment `collectCoverageFrom: ['src/**/*.{ts,tsx}']` then tests execution time slightly decreases (about 0.8 second). That happens because now a coverage report doesn't include modules without tests for them. Frankly, I expected the opposite, I supposed that it would take much more time and would include every folder in my project including "node_modules"  

![test_coverage_without_collectCoverageFrom](https://user-images.githubusercontent.com/23404373/203355285-98546c75-7079-4508-a310-cfa40c4d3262.jpg)